app-creator/deserializer.js

```diff
@@ -0,0 +1,438 @@
+    var map = obj.map;
+    var node = obj.node;
+
+    app.widgetInterface[nodeObj.id] = obj;
+
+    for (var i = 0; i < nodeObj.children.length; i++) {
+      var widget = nodeObj.children[i];
+      if (map !== undefined) {
+        map.add(helper(widgetTree[widget]));
+      } else {
+        node.widgets().add(helper(widgetTree[widget]));
+      }
+    }
+
+    return node;
+  }
+
+  var widgetTree = JSON.parse(widgetTreeJSON);
+  return helper(widgetTree['panel-template-0']);
+};
+
+/**
+ * Takes in a node object containing meta data about a node and returns the created ui element.
+ */
+app.createUIElement = function (nodeObj) {
+  var type = nodeObj.id.slice(0, nodeObj.id.indexOf('-'));
+  var filteredStyles = app.filterStyleObject(nodeObj.style);
+  switch (type) {
+    case 'panel':
+      return app.createPanelElement(nodeObj, filteredStyles);
+    case 'map':
+      return app.createMapElement(nodeObj, filteredStyles);
+    case 'label':
+      return app.createLabelElement(nodeObj, filteredStyles);
+    case 'button':
+      return app.createButtonElement(nodeObj, filteredStyles);
+    case 'checkbox':
+      return app.createCheckboxElement(nodeObj, filteredStyles);
+    case 'textbox':
+      return app.createTextboxElement(nodeObj, filteredStyles);
+    case 'select':
+      return app.createSelectElement(nodeObj, filteredStyles);
+    case 'slider':
+      return app.createSliderElement(nodeObj, filteredStyles);
+    case 'chart':
+      return app.createChartElement(nodeObj, filteredStyles);
+    default:
+      return { node: null };
+  }
+};
+
+/**
+ * Helper function for creating slider elements.
+ */
+app.createSliderElement = function (obj, style) {
+  var uniqueAttributes = obj.uniqueAttributes;
+
+  var min = parseFloat(uniqueAttributes.min);
+  min = isNaN(min) ? 0 : min;
+
+  var max = parseFloat(uniqueAttributes.max);
+  max = isNaN(max) ? 1 : max;
+
+  var step = parseFloat(uniqueAttributes.step);
+  step = isNaN(step) ? 0.01 : step;
+
+  var value = parseFloat(uniqueAttributes.value);
+  value = isNaN(value) ? 0 : value;
+
+  var direction = uniqueAttributes.direction;
+
+  var disabled = uniqueAttributes.disabled === 'true';
+
+  var slider = ui.Slider({
+    min: min,
+    max: max,
+    value: value,
+    step: step,
+    direction: direction,
+    disabled: disabled,
+    style: style,
+  });
+
+  return { node: slider };
+};
+
+/**
+ * Helper function for creating chart elements.
+ */
+app.createChartElement = function createChartElement(obj, style) {
+  var uniqueAttributes = obj.uniqueAttributes;
+
+  var dataTable = uniqueAttributes.dataTable;
+
+  var chartType = uniqueAttributes.chartType.replace(/\s/g, '');
+
+  var title = uniqueAttributes.title;
+
+  var color = uniqueAttributes.color;
+  color = color.split(',');
+  for (var i = 0; i < color.length; i++) {
+    color[i] = color[i].trim();
+  }
+
+  var threeD = uniqueAttributes['3D'];
+
+  var downloadable = uniqueAttributes.downloadable === 'true';
+
```

Comment:
Right now I'm just adding a placeholder for the dataTable. I'll be changing the way we add columns and rows in a following PR.